.github/workflows/ci.yml

```diff
@@ -90,6 +90,109 @@ jobs:
         command: test
         args: --all  --target ${{ matrix.target }}
 
+  build_release:
+    name: Build release binaries
+    runs-on: ${{ matrix.runner }}
+    continue-on-error: false
+    needs: build_and_test_nix
+    # if: ${{ always() && github.ref_name=='main' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        name: [ubuntu-latest, ubuntu-arm-latest, macOS-latest, macOS-arm-latest]
+        rust: [stable]
+        experimental: [false]
+        include:
+          - name: ubuntu-arm-latest
+            os: ubuntu-latest
+            release-os: linux
+            release-arch: aarch64
+            protoc-arch: linux-aarch_64
+            runner: [self-hosted, linux, ARM64]
+          - name: ubuntu-latest
+            os: ubuntu-latest
+            release-os: linux
+            release-arch: amd64
+            protoc-arch: linux-x86_64
+            runner: [self-hosted, linux, X64]
+          - name: macOS-latest
+            os: macOS-latest
+            release-os: darwin
+            release-arch: x86_64
+            protoc-arch: osx-x86_64
+            runner: [self-hosted, macOS, X64]
+          - name: macOS-arm-latest
+            os: macOS-latest
+            release-os: darwin
+            release-arch: aarch64
+            protoc-arch: osx-aarch_64
+            runner: [self-hosted, macOS, ARM64]
+          # - os: windows-latest
+          #   release-os: windows
+          #   release-arch: amd64
+          #   runner: [windows-latest]
+    env:
+      RUST_BACKTRACE: full
+      # Force not building debuginfo to save space on disk.
+      RUSTFLAGS: "-C debuginfo=0"
```

Comment:
the disk space thing is a different issue with GH runners, no longer an issue.
on the debug info I'm for keeping it in for now as we don't really have large binary sizes and it will be handy in the nearer future.